src/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from 'react';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+
+import { InputGroup, Button } from '../../index';
+import AutoCompleteInput from './AutoCompleteInput';
+import AutoCompleteItems from './AutoCompleteItems';
+import { KEY_CODES } from '../../common/helpers';
+
+export const getActiveItems = items =>
+  items
+    .filter(
+      ({ disabled, type }) => !disabled && !['header', 'divider'].includes(type)
+    )
+    .map(({ text }) => text);
+
+class AutoComplete extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: ''
+    };
+  }
+
+  render() {
+    const {
+      onSearch,
+      labelText,
+      onInputUpdate,
+      items,
+      actionText,
+      ...rest
+    } = this.props;
+
+    const activeItems = getActiveItems(items);
+
+    return (
+      <Downshift
+        onStateChange={({ inputValue, ...rest }) => {
+          if (typeof inputValue === 'string') {
+            onInputUpdate(inputValue);
+            this.setState({ inputValue });
+          }
+        }}
+        defaultHighlightedIndex={0}
+        selectedItem={this.state.inputValue}
+        {...rest}
+      >
+        {({
+          getInputProps,
+          getItemProps,
+          getLabelProps,
+          isOpen,
+          inputValue,
+          highlightedIndex,
+          getRootProps,
+          selectedItem,
+          selectItem
+        }) => (
+          <div>
+            {labelText && <label {...getLabelProps()}>{labelText}</label>}
+            <InputGroup>
+              <AutoCompleteInput
+                onKeyPress={e => {
+                  switch (e.keyCode) {
+                    case KEY_CODES.TAB_KEY:
+                      if (isOpen && activeItems[highlightedIndex]) {
+                        selectItem(activeItems[highlightedIndex]);
+                        e.preventDefault();
+                      }
+
+                      break;
+
+                    case KEY_CODES.ENTER_KEY:
+                      if (!isOpen || !activeItems[highlightedIndex]) {
+                        onSearch(this.state.inputValue);
+                        e.preventDefault();
+                      }
+
+                      break;
+
+                    default:
+                      break;
+                  }
+                }}
+                passedProps={getInputProps()}
+              />
+              <InputGroup.Button>
+                <Button onClick={() => onSearch(inputValue)}>
+                  {actionText}
+                </Button>
+              </InputGroup.Button>
+            </InputGroup>
+
+            {isOpen && items.length ? (
```

Comment:
My suggestion here:
```jsx
const shouldShowItems = isOpen && items.length && items.length > 0;
const autoCompleteItemsProps = { ... };

{shouldShowItems && <AutoCompleteItems {...autoCompleteItemsProps} />}
```

Reply:
😍  this pattern. I think we can remove the `items.length` check since it's a require array. The `items.length > 0` should be sufficient.